translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,294 @@
+
+    # Relationship
+    properties['relationship'] = {}
+
+    # lanes
+    properties['lanes'] = []
+
+    # beginning_cross_street
+    properties['beginning_cross_street'] = ""
+
+    # beginning_cross_street
+    properties['ending_cross_street'] = ""
+
+    # event status
+    properties['event_status'] = get_event_status(event['header']['start_timestamp'], event['header'].get('end_timestamp'))
+
+    # type_of_work
+    # maintenance, minor-road-defect-repair, roadside-work, overhead-work, below-road-work, barrier-work, surface-work, painting, roadway-relocation, roadway-creation
+    properties['types_of_work'] = []
+
+    # restrictions
+    properties['restrictions'] = [] 
+
+    # description
+    properties['description'] = event['header']['description']
+
+    # creation_date
+    properties['creation_date'] = reformat_datetime(event['source']['collection_timestamp'])
+
+    # update_date
+    properties['update_date'] = reformat_datetime(alert['rtdh_timestamp'])
+
+    feature = {}
+    feature['type'] = "Feature"
+    feature['properties'] = properties
+    feature['geometry'] = geometry
+
+    return feature
+
+#function to validate the alert
+def validate_alert(alert):
+    
+    if not alert or (type(alert) != dict and type(alert) != OrderedDict):
+        logging.warning('alert is empty or has invalid type')
+        return False
+    
+    event = alert.get('event', {})
+    polyline = event.get('geometry')
+    coords = parse_polyline(polyline)
+    street = event.get('detail', {}).get('road_name')
+    
+    header = event.get('header', {})
+    starttime = header.get('start_timestamp')
+    endtime = header.get('end_timestamp', 0)
+    description = header.get('description')
+    direction = event.get('detail', {}).get('direction')
+   
+    required_fields = [ polyline, coords, street, starttime, description, direction]
+    for field in required_fields:
+        if not field:
+            logging.warning(f'Invalid event with event id = {alert.get("rtdh_message_id")}. not all required fields are present')
+            return False
+            
+        
+        if type(starttime) != int or type(endtime) != int:
+            logging.warning(f'Invalid event with id = {alert.get("rtdh_message_id")}. Invalid datetime format')
+            return False
+    
+    return True
+
+def reformat_datetime(datetime_string):
+    if not datetime_string:
+        return ''
+    elif type(datetime_string) == str:
+        if re.match('^-?([0-9]*[.])?[0-9]+$', datetime_string):
+            datetime_string = float(datetime_string)
+        else:
+            return ''
+    time = datetime.fromtimestamp(datetime_string)
+    wzdx_format_datetime = time.astimezone(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+    return wzdx_format_datetime
+
+# Add ids to message
+#### This function may fail if some optional fields are not present (lanes, types_of_work, relationship, ...)
+def add_ids(message, add_ids):
```

Comment:
Do we need the `add_ids` argument? Seems like we always call this function with it set to True

Reply:
i deleted the add_ids argument